powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java

```diff
@@ -373,13 +375,33 @@ private void logError(TppEngineClientException ex) {
         }
     }
 
+    private static TppEngineClientException createTppEngineClientException(final RestClientException e) {
+        return new TppEngineClientException(e, resolveTppEngineError(e));
+    }
+
+    private static TppEngineError resolveTppEngineError(final RestClientException e) {
+        // TODO (racansky, 2022-12-02) workaround until https://github.com/wultra/lime-java-core/issues/32
+        if (e.getErrorResponse() == null || e.getErrorResponse().getResponseObject() == null) {
+            try {
+                return new ObjectMapper().readValue(e.getResponse(), new TypeReference<ObjectResponse<TppEngineError>>(){})
+                        .getResponseObject();
+            } catch (JsonProcessingException ex2) {
+                logger.debug("Problem to deserialize error response", ex2);
+                return new TppEngineError(resolveErrorCode(e), e.getMessage());
+            }
+        }
+
+        final Error error = e.getErrorResponse().getResponseObject();
+        return new TppEngineError(error.getCode(), error.getMessage());
+    }
+
     /**
      * Resolve error code based on HTTP status code from REST client exception.
      */
-    private String resolveErrorCode(RestClientException ex) {
+    private static String resolveErrorCode(RestClientException ex) {
```

Comment:
Just out of curiosity, is there a reason for the `static` modifier?

Reply:
But all code runs inside of an instance, so what is the point? It doesn't matter, just it is a bit misleading to me, but OK.

Reply:
IMHO once the method has been made static, it is clear that it can be moved out of the class.